File: qc-application/src/main/java/com/eager/questioncloud/application/exception/ErrorResponse.java
package com.eager.questioncloud.application.exception;

import com.eager.questioncloud.core.exception.CoreException;
import lombok.Builder;
import lombok.Getter;
import org.springframework.http.ResponseEntity;

@Builder
@Getter
public class ErrorResponse {
    private int status;
    private String message;

    public static ResponseEntity<ErrorResponse> toResponse(CoreException e) {
        return ResponseEntity
            .status(e.getError().getHttpStatus())
            .body(ErrorResponse
                .builder()
                .status(e.getError().getHttpStatus())
                .message(e.getError().getMessage())
                .build());
    }
}

File: qc-application/src/main/java/com/eager/questioncloud/application/exception/ErrorResponse.kt
package com.eager.questioncloud.application.exception

import com.eager.questioncloud.core.exception.CoreException
import org.springframework.http.ResponseEntity

class ErrorResponse(
    val status: Int,
    val message: String,
) {
    companion object {
        fun toResponse(e: CoreException): ResponseEntity<ErrorResponse> {
            return ResponseEntity
                .status(e.error.httpStatus)
                .body(ErrorResponse(e.error.httpStatus, e.error.message))
        }
    }
}